testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+        )
+        cmd = cmd + ' '.join(principal for principal in self.principals)
+        try:
+            sdk_cmd.run_cli(cmd)
+        except Exception as e:
+            raise(e)
+
+        _copy_file_to_localhost(self)
+
+    def __create_and_upload_secret(self):
+        """
+        This method base64 encodes the keytab file and creates a secret with this encoded content so the
+        tasks can fetch it.
+        """
+        log.info("Creating and uploading the keytab file to the secret store")
+
+        cmd = "package install --yes --cli dcos-enterprise-cli"
+        try:
+            sdk_cmd.run_cli(cmd)
+        except dcos.errors.DCOSException as e:
+            raise RuntimeError("Failed to install the dcos-enterprise-cli: {}".format(repr(e)))
+
+        try:
+            base64_encode_cmd = "base64 -w 0 {source} > {destination}".format(
+                source=os.path.join(self.temp_working_dir, self.keytab_file_name),
+                destination=os.path.join(self.temp_working_dir, self.base64_encoded_keytab_file_name)
+            )
+            run(base64_encode_cmd, shell=True)
+        except Exception as e:
+            raise Exception("Failed to base64-encode the keytab file: {}".format(repr(e)))
+
+        self.keytab_secret_path = "{}_keytab".format(DCOS_BASE64_PREFIX)
+
+        # TODO: check if a keytab secret of same name already exists
+        create_secret_cmd = "security secrets create {keytab_secret_path} --value-file {encoded_keytab_path}".format(
+            keytab_secret_path=self.keytab_secret_path,
+            encoded_keytab_path=os.path.join(self.temp_working_dir, self.base64_encoded_keytab_file_name)
+        )
+        try:
+            sdk_cmd.run_cli(create_secret_cmd)
+        except RuntimeError as e:
+            raise RuntimeError("Failed to create secret for the base64-encoded keytab file: {}".format(repr(e)))
+
+        log.info("Successfully uploaded a base64-encoded keytab file to the secret store")
+
+    def finalize_environment(self):
+        """
+        Once the principals have been added, the rest of the environment setup does not ask for more info and can be
+        automated, hence this method.
+        """
+        self.__create_and_fetch_keytab()
+        self.__create_and_upload_secret()
+
+    def get_host(self):
+        return self.kdc_fqdn
+
+    def get_port(self):
+        return str(self.kdc_port)
+
+    def get_keytab_path(self):
+        return self.keytab_secret_path
+
+    def cleanup(self):
+        sdk_marathon.destroy_app(KERBEROS_APP_ID)
+        if os.path.exists(self.temp_working_dir):
+            shutil.rmtree(self.temp_working_dir)
+
+        delete_secret_cmd = "security secrets delete {}".format(self.keytab_secret_path)
+        try:
+            sdk_cmd.run_cli(delete_secret_cmd)
+        except RuntimeError as e:
+            raise RuntimeError("Failed to delete secret for the base64-encoded keytab file: {}".format(repr(e)))
```

Comment:
As far as I remember, in the hello-world secret tests, we are a bit more flexible in terms of the success of this command. If, for example, the secret was not created due to an error, we would not expect this to pass.

Reply:
For later: We could add an `sdk_secrets` module that wraps this kind of operation for us.

Reply:
Yeah might be worth considering. Add this thought as a comment.